### PR TITLE
Add hurricane tracker code

### PR DESCRIPTION
**Pull request by caternuson**
Here it is! Pull request like a hurricane.

### CODE REVIEW

File: PyPortal_Hurricane_Tracker/hurricane_tracker.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    # clear out existing icons
+    while len(storm_icons):
+        _ = storm_icons.pop()
+
+    # get latest storm data
+    try:
+        storm_data = pyportal.fetch()
+    except RuntimeError:
+        return
+    print("Number of storms:", len(storm_data))
```
Comment (jerryneedell) [inline PyPortal_Hurricane_Tracker/hurricane_tracker.py:73]: This prints # of storms but later only storms in area are displayed

File: PyPortal_Hurricane_Tracker/hurricane_tracker.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        if len(storm_icons) >= MAX_STORMS:
+            continue
+        # get lat/lon
+        lat = storm["latitudeNumeric"]
+        lon = storm["longitudeNumeric"]
+        # check if on map
+        if (
+            not LAT_RANGE[0] >= lat >= LAT_RANGE[1]
+            or not LON_RANGE[0] <= lon <= LON_RANGE[1]
+        ):
```
Comment (jerryneedell) [inline PyPortal_Hurricane_Tracker/hurricane_tracker.py:87]: Do you want to print something if storms out of area. I was confused when it said there were two storms found, but only printed information on one.

File: PyPortal_Hurricane_Tracker/hurricane_tracker.py (hunk 1 above)
Comment (caternuson) [reply PyPortal_Hurricane_Tracker/hurricane_tracker.py:73]: In general, all the serial prints are debug info. For this one, was helpful to know actual storm count in raw JSON.

File: PyPortal_Hurricane_Tracker/hurricane_tracker.py (hunk 2 above)
Comment (caternuson) [reply PyPortal_Hurricane_Tracker/hurricane_tracker.py:87]: Same here. If no storms end up being within the map area, then there's another debug print that should show up.

**Comment by caternuson** [comment]
Awesome, thanks for testing it. Good to know it works :)

File: PyPortal_Hurricane_Tracker/hurricane_tracker.py (hunk 1 above)
Comment (jerryneedell) [reply PyPortal_Hurricane_Tracker/hurricane_tracker.py:73]: This is fine.. and useful.

File: PyPortal_Hurricane_Tracker/hurricane_tracker.py (hunk 2 above)
Comment (jerryneedell) [reply PyPortal_Hurricane_Tracker/hurricane_tracker.py:87]: As I said, no big deal -- it just took me a few minutes to figure out why it said there were 2 storms and only showed 1.
I saw that if there are none in range it prints a message.
I'm fine with it.

**Comment by jerryneedell** [comment]
Looks like the map is only for Atlantic  -- Is that going to be specified in the Guide -- or will it be expanded for other regions as well.

**Comment by caternuson** [comment]
Yep. Just Atlantic for now. Maybe add other regions later. But also - that would make for an excellent user update. Use the provided Atlantic as an example and extent from there.

The NOAA JSON source covers Atlantic, East Pacific, and Central Pacific. So those regions could be added fairly easily.

The other missing bit is a map for a Titano. Simple resize, but didn't have the hardware to test.

**Comment by jerryneedell** [comment]
> Yep. Just Atlantic for now. Maybe add other regions later. But also - that would make for an excellent user update. Use the provided Atlantic as an example and extend from there.
> 
> The NOAA JSON source covers Atlantic, East Pacific, and Central Pacific. So those regions could be added fairly easily.
> https://nws.weather.gov/products/PDD/PDD_ExpTropicalCycloneJSONFile_2019.pdf
> 
> The other missing bit is a map for a Titano. Simple resize, but didn't have the hardware to test.

ooh -- I have a Titano -- I suppose I should learn how to do the resize...

**Comment by caternuson** [comment]
Oh, they're different aspect ratios:

- Pynt / Original: 320x240 = 4:3
- Titano : 480x320 = 3:2

so actually a little more than just a resize.

**Pull request merged**